Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/risk/PresenceTracingRiskRepository.kt

```diff
@@ -51,37 +61,70 @@ class PresenceTracingRiskRepository @Inject constructor(
             presenceTracingRiskCalculator.calculateAggregatedRiskPerDay(it)
         }
 
-    suspend fun replaceAllMatches(list: List<CheckInWarningOverlap>) {
-        deleteAllMatches()
-        addAll(list)
+    internal suspend fun reportSuccessfulCalculation(list: List<CheckInWarningOverlap>) {
+        traceTimeIntervalMatchDao.insert(list.map { it.toEntity() })
+        val fifteenDaysAgo = timeStamper.nowUTC.minus(Days.days(15).toStandardDuration()).toLocalDateUtc()
```

Comment:
(fifteenDaysAgo) create a fun for this, that you dont have to repeat it elsewhere/below